scripts/build_shapes.py

```diff
@@ -805,15 +875,26 @@ def gadm(
     out_logging = snakemake.config["build_shape_options"]["out_logging"]
     year = snakemake.config["build_shape_options"]["year"]
     nprocesses = snakemake.config["build_shape_options"]["nprocesses"]
+    contended_areas = snakemake.config["build_shape_options"]["contended_areas"]
     EEZ_gpkg = snakemake.input["eez"]
     worldpop_method = snakemake.config["build_shape_options"]["worldpop_method"]
     gdp_method = snakemake.config["build_shape_options"]["gdp_method"]
     geo_crs = snakemake.config["crs"]["geo_crs"]
     distance_crs = snakemake.config["crs"]["distance_crs"]
 
-    country_shapes = countries(countries_list, geo_crs, update, out_logging)
+    keep_all_areas = False if contended_areas == "drop" else True
 
-    country_shapes.reset_index().to_file(snakemake.output.country_shapes)
+    country_shapes = countries(
+        countries_list, geo_crs, update, out_logging, keep_all_areas=True
+    )
+
+    # there may be "holes" in the countries geometry which cause troubles along the workflow
+    # e.g. that is the case for enclaves like Dahagram–Angarpota for IN/BD
+    country_shapes_valid = (
+        country_shapes.apply(lambda x: make_valid(x) if not x.is_valid else x)
```

Comment:
This apply and filter may be moved into the countries function so to clean the main+

Reply:
Agree and done